lib/packs/private/pack_relationship_analyzer.rb

```diff
@@ -99,7 +37,7 @@ def self.list_top_dependency_violations(pack_name, limit)
 
         # TODO: This is a copy of the implementation above. We may want to refactor out this implementation detail before making changes that apply to both.
         all_packages.each do |client_package|
-          client_package.violations.select(&:dependency?).each do |violation|
+          client_package.violations.select { _1.type == type }.each do |violation|
```

Comment:
This is the only real change that was needed in order for this code to be usable not only for dependency but also for privacy and architecture todos/violations.